src/operator/matrix_op-inl.h

```diff
@@ -144,14 +231,180 @@ inline TShape DotShape(const TShape& lshape,
 }
 
 
+struct SimpleCropParam : public dmlc::Parameter<SimpleCropParam> {
+  TShape begin, end;
+  DMLC_DECLARE_PARAMETER(SimpleCropParam) {
+    DMLC_DECLARE_FIELD(begin)
+    .describe("starting coordinates");
+    DMLC_DECLARE_FIELD(end)
+    .describe("ending coordinates");
+  }
+};
+
+// matrix crop
+template<typename xpu>
+void Crop(const TBlob &src,
+          const EnvArguments& env,
+          TBlob *ret,
+          OpReqType req,
+          RunContext ctx) {
+  using namespace mshadow;
+  using namespace mshadow::expr;
+  SimpleCropParam param;
+  param.Init(env.kwargs);
+  CHECK_EQ(src.type_flag_, ret->type_flag_);
+  Stream<xpu> *s = ctx.get_stream<xpu>();
+  MSHADOW_TYPE_SWITCH(ret->type_flag_, DType, {
+    switch (src.shape_.ndim()) {
+     case 0:
+      break;
+     case 1: {
+      Tensor<xpu, 1, DType> in = src.get<xpu, 1, DType>(s);
+      Tensor<xpu, 1, DType> out = ret->get<xpu, 1, DType>(s);
+      out = slice(in, param.begin.get<1>(), param.end.get<1>());
+      break;
+     }
+     case 2: {
+      Tensor<xpu, 2, DType> in = src.get<xpu, 2, DType>(s);
+      Tensor<xpu, 2, DType> out = ret->get<xpu, 2, DType>(s);
+      out = slice(in, param.begin.get<2>(), param.end.get<2>());
+      break;
+     }
+     case 3: {
+      Tensor<xpu, 3, DType> in = src.get<xpu, 3, DType>(s);
+      Tensor<xpu, 3, DType> out = ret->get<xpu, 3, DType>(s);
+      out = slice(in, param.begin.get<3>(), param.end.get<3>());
+      break;
+     }
+     case 4: {
+      Tensor<xpu, 4, DType> in = src.get<xpu, 4, DType>(s);
+      Tensor<xpu, 4, DType> out = ret->get<xpu, 4, DType>(s);
+      out = slice(in, param.begin.get<4>(), param.end.get<4>());
+      break;
+     }
+     case 5: {
+      Tensor<xpu, 5, DType> in = src.get<xpu, 5, DType>(s);
+      Tensor<xpu, 5, DType> out = ret->get<xpu, 5, DType>(s);
+      out = slice(in, param.begin.get<5>(), param.end.get<5>());
+      break;
+     }
+     default:
+      LOG(FATAL) << "crop supports at most 5 dimensions";
+      break;
+    }
```

Comment:
For these kind of switches, would it be better to use macros to simplify the code like the `MXNET_RANGE_SWITCH` in https://github.com/dmlc/mxnet/pull/2292